src/index.ts

```diff
@@ -0,0 +1,17 @@
+export class Challenge {
+    num : number;
+    constructor(){
+    }
+    public buzz(i:number){
+        return new Promise(function(resolve, reject){
+            if(i%3==0 && i%5==0)
+                resolve("fizzbuzz");
+            if(i%3==0)
+                resolve("fizz");
+            if(i%5==0)
+                resolve("buzz");
+            else
```

Comment:
In this case (without `else if`) this refers only to last `if`. Maybe this wasn't what you had in mind ;)